lib/src/container/store.rs

```diff
@@ -441,8 +441,11 @@ impl ImageImporter {
         imgref: &OstreeImageReference,
         mut config: ImageProxyConfig,
     ) -> Result<Self> {
-        // Apply our defaults to the proxy config
-        merge_default_container_proxy_opts(&mut config)?;
+        // Pulling from container-storage doesnt need to drop privileges
+        if imgref.imgref.transport != Transport::ContainerStorage {
+            // Apply our defaults to the proxy config
+            merge_default_container_proxy_opts(&mut config)?;
```

Comment:
Hmm.  I think we may instead want to call `merge_default_container_proxy_opts_with_isolation(&mut config, None)` ...the difference here boils down to the authfile handling, which is also irrelevant when pulling from containers-storage...but in the future someone might add *other* config handling there that might be relevant.

Reply:
IOW more like:

```
        if imgref.imgref.transport != Transport::ContainerStorage {
            // Apply our defaults to the proxy config
            merge_default_container_proxy_opts(&mut config)?;
        } else {
            // When fetching from containers-storage, we can't drop privileges since
            // we may need them to read files.
            merge_default_container_proxy_opts_with_isolation(&mut config, None)?;
        }
```

or so?

Reply:
Agree that this is a cleaner approach